launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java

```diff
@@ -315,7 +316,7 @@ static int javaMajorVersion(String javaVersion) {
    */
   static String findJarsDir(String sparkHome, String scalaVersion, boolean failIfNotFound) {
     // TODO: change to the correct directory once the assembly build is changed.
-    File libdir = new File(sparkHome, "jars");
+    File libdir = new File(getJarsDir(sparkHome));
     if (!libdir.isDirectory()) {
       libdir = new File(sparkHome, String.format("assembly/target/scala-%s/jars", scalaVersion));
       if (!libdir.isDirectory()) {
```

Comment:
Ur, this PR seems to cause a misleading error message when the user give a wrong `SPARK_JARS_DIR`.

In case of a wrong `SPARK_JARS_DIR`, the error message should give the user-given `SPARK_JARS_DIR` instead of `assembly/target/scala-%s/jars`. Could you check this once more, @marblejenka ?

Also, it would be great if we can have a negative test coverage for the above wrong `SPARK_JARS_DIR`.